impeller/renderer/backend/vulkan/render_pass_vk.cc

```diff
@@ -592,27 +592,42 @@ bool RenderPassVK::OnEncodeCommands(const Context& context) const {
 
   const auto& target_size = render_target_.GetRenderTargetSize();
 
-  auto render_pass = CreateVKRenderPass(vk_context);
-  if (!render_pass) {
-    VALIDATION_LOG << "Could not create renderpass.";
-    return false;
-  }
+  RingBufferItem<RenderPassFrameBufferPair> render_pass_frame_buffer_pair =
+      vk_context.GetRenderPassFrameBufferPair();
 
-  auto framebuffer = CreateVKFramebuffer(vk_context, *render_pass);
-  if (!framebuffer) {
-    VALIDATION_LOG << "Could not create framebuffer.";
-    return false;
+  if (!render_pass_frame_buffer_pair.GetValue()) {
+    auto render_pass = CreateVKRenderPass(vk_context);
+    if (!render_pass) {
+      VALIDATION_LOG << "Could not create renderpass.";
+      return false;
+    }
+
+    auto framebuffer = CreateVKFramebuffer(vk_context, *render_pass);
+    if (!framebuffer) {
+      VALIDATION_LOG << "Could not create framebuffer.";
+      return false;
+    }
+
+    RenderPassFrameBufferPair pair = RenderPassFrameBufferPair{
+        .render_pass = render_pass,
+        .framebuffer = framebuffer,
+    };
+
+    render_pass_frame_buffer_pair.GetValue() = std::move(pair);
   }
 
-  if (!encoder->Track(framebuffer) || !encoder->Track(render_pass)) {
+  if (!encoder->Track(
```

Comment:
@jonahwilliams right here we are making sure it isn't placed back into the pool until the frame is done.